extends OptionButton


var langs
const langnames = ["Ð‘ÐµÐ»Ð°Ñ€ÑƒÑÐºÐ°Ñ", "ä¸­æ–‡", "English", "Ð ÑƒÑÑÐºÐ¸Ð¹", "EspaÃ±ol", "Suomi", "Bielaruskaja lacinka", "Polska", "ðŸ¦² (emoji)",]
const langtooltips = [
	"ÐÐ´Ð½Ð¾ Ð· Ð³Ð°Ð»Ð¾ÑžÐ½Ñ‹Ñ… Ð½Ð¾Ð²Ð°ÑžÐ²ÑÐ´Ð·ÐµÐ½Ð½ÑÑž ZR 2.0 Ð³ÑÑ‚Ð° Ð›ÐÐšÐÐ›Ð†Ð—ÐÐ¦Ð«Ð¯! Ð’Ñ‹ Ð¼Ð¾Ð¶Ð°Ñ†Ðµ Ð·Ð¼Ð°Ð³Ð°Ñ†Ñ†Ð° Ð· Ð·Ð¾Ð¼Ð±Ñ– Ð±ÐµÐ· Ð¼Ð¾ÑžÐ½Ð°Ð³Ð° Ð±Ð°Ñ€'ÐµÑ€Ð°! Ð£Ð Ð!",
	"ZR 2.0 çš„ä¸»è¦æ–°åŠŸèƒ½ä¹‹ä¸€æ˜¯æœ¬åœ°åŒ–ï¼æ‚¨å¯ä»¥åœ¨æ²¡æœ‰è¯­è¨€éšœç¢çš„æƒ…å†µä¸‹ä¸Žåƒµå°¸æˆ˜æ–—ï¼è€¶",
	"One of the main new features of ZR 2.0 is LOCALIZATION! You can now fight zombies without a language barrier! YAY!",
	"ÐžÐ´Ð½Ð¾ Ð¸Ð· Ð³Ð»Ð°Ð²Ð½Ñ‹Ñ… Ð½Ð¾Ð²Ð¾Ð²ÐµÐ´ÐµÐ½Ð½Ð¸Ð¹ ZR 2.0 ÑÑ‚Ð¾ Ð›ÐžÐšÐÐ›Ð˜Ð—ÐÐ¦Ð˜Ð¯! Ð’Ñ‹ Ð¼Ð¾Ð¶ÐµÑ‚Ðµ ÑÑ€Ð°Ð¶Ð°Ñ‚ÑŒÑÑ Ñ Ð·Ð¾Ð¼Ð±Ð¸ Ð±ÐµÐ· ÑÐ·Ñ‹ÐºÐ¾Ð²Ð¾Ð³Ð¾ Ð±Ð°Ñ€ÑŒÐµÑ€Ð°! Ð£Ð Ð!",
	"Una de las principales novedades de ZR 2.0 es Â¡La LOCALIZACIÃ“N! Â¡PodrÃ¡s luchar contra zombis sin la barrera del idioma! Â¡YAY!",
	"Yksi ZR 2.0:n tÃ¤rkeimmistÃ¤ innovaatioista on LOKALISAATIO! Voit taistella zombeja vastaan â€‹â€‹ilman kielimuuria! HURRAA!",
	"Adno z haloÅ­nych novaÅ­viadzieÅ„niaÅ­ ZR 2.0 heta LAKALIZACYJa! Vy moÅ¾acie zmahacca z zombi biez moÅ­naha barjera! URA!",
	"PL PLACEHOLDER",
	"1âƒ£ ðŸ‘‘ ðŸ’¡ ðŸ’§ 2âƒ£.0âƒ£ ðŸ”— â— ðŸ‘† ðŸ’ª âš” ðŸ§Ÿâ€â™‚ ðŸµ ðŸ‡ªðŸ‡¸ ðŸš§ â— ðŸ‘ â—",
]

func _ready() -> void:
	
	langs = TranslationServer.get_loaded_locales()
	print(langs)
	for n in langs.size():
		print(n)
		add_item(langnames[n], -1)
		var sus = n + 1
		set_item_tooltip(n, langtooltips[n])
	
	selected = TranslationServer.get_loaded_locales().find(TranslationServer.get_locale(), 0)
	
func _on_item_selected(index: int) -> void:
	TranslationServer.set_locale(langs[index])
	
	print(langs[index])
